app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1157,8 +1161,11 @@ private void handleFormLoadCompletion(AndroidFormController fc) {
         uiController.refreshView();
         FormNavigationUI.updateNavigationCues(this, mFormController, uiController.questionsView);
         if (isRestartAfterSessionExpiration) {
-            Toast.makeText(this,
-                    Localization.get("form.entry.restart.after.expiration"), Toast.LENGTH_LONG).show();
+            String localeKey =
+                    (fc.getInterruptedFormState() == null
```

Comment:
can `fc.getInterruptedFormState() == null` be ever true here given `isRestartAfterSessionExpiration` is already true ?

Reply:
For backward compatibility, we might load a previously saved session using an older version of CC, where the InterruptedFormState was not saved.

Reply:
makes sense, can we make a comment  there that the null check is to support backward compatibility and should not be removed.

Reply:
[a27ab98](https://github.com/dimagi/commcare-android/pull/2858/commits/a27ab98c1f70f7afd060af5cf31ddd25f8e3def4)